Doc/Manual/Fortran.html

```diff
@@ -0,0 +1,2275 @@
+</p>
+
+
+<p>
+Each function in SWIG has a unique &ldquo;symbolic name&rdquo; or <em>symname</em> bound to it. The <em>symname</em> must be compatible with C linkage, and thus namespaces, templates, and overloads are incorporated into the symname, but a symname is often just the same as the bare function name. <!--
+For example, a function 
+```c++
+namespace foo {
+template<typename T>
+void bar(T arg);
+}
+```
+and instantiated with
+```swig
+%template(bar_dbl) foo::bar<double>;
+```
+might a symbolic name `bar_dbl
+-->
+</p>
+
+
+<p>
+SWIG will generate a C-linkage wrapper function in the C++ file named <code>_wrap_$symname</code>, where <code>$symname</code> is replaced with the symname. This wrapper function is responsible for converting the function&rsquo;s arguments and return value to and from Fortran-compatible datatypes and calling the C++ function. It also implements other optional features such as exception handling.
+</p>
+
+
+<p>
+SWIG then creates an <code>interface</code> declaration <code>swigc_$symname</code>, with <code>bind(C, name='_wrap_$symname')</code>. interface statement that aliases that wrapper symbol to <code>swigc_$symname</code> will be generated in the Fortran interface module.
+</p>
+
+
+<p>
+In the Fortran module, SWIG generates a public procedure <code>$symname</code> that translates native Fortran data types to and from the C interface datatypes. This interface, and not the <code>swigc_$symname</code> bound function, is the one used by Fortran application codes.
+</p>
+
+<H3><a name="Fortran_function_overloading">Function overloading</a></H3>
+
+<p>
+There is an important exception to the naming scheme described above: function overloading, when two or more free functions share a name but have different arguments. For each overloaded function signature, SWIG generates a private procedure with a unique symname. These procedures are then combined under a <em>separate module procedure</em> that is given a public interface with the original symbolic name. For example, an overloaded free function <code>myfunc</code> in C++ will generate two private procedures and add an interface to the module specification:
+</p>
+
+
+<div class="targetlang"><pre><code> public :: myfunc
+ interface myfunc
+  module procedure myfunc__SWIG_0, myfunc__SWIG_1
+ end interface</code></pre></div>
+
+
+<p>
+It should be noted that a function that returns <code>void</code> cannot be overloaded with a function that returns anything else: generic interfaces must be either all subroutines or all functions. Use SWIG&rsquo;s <code>%ignore</code> statement to hide one or the other:
+</p>
```

Comment:
This type of overload doesn't compile (Fortran compile) when wrapped. SWIG aims to generate code that will compile by default. 

If this type of overload occurs, one of the functions should be ignored by default with a
Fortran specific warning. I suggest ignoring all the functions with void returns as a
rule if there are overloaded functions containing a mix of void/non-void return.

Reply:
For future reference, overload_method.i tests this kind of overload and I see it is currently in the FAILING_CPP_TESTS in Examples/test-suite/fortran/Makefile.in. @sethrj, I suggest you add this under your list of known issues.

Reply:
This behavior has been fixed and the documentation updated accordingly.